src/ApcuCache.php

```diff
@@ -86,12 +109,81 @@ private function normalizeTtl($ttl): ?int
     }
 
     /**
-     * Converts iterable to array
-     * @param iterable $iterable
+     * Converts iterable to array. If provided value is not iterable it throws an InvalidArgumentException
+     * @param $iterable
      * @return array
      */
-    private function iterableToArray(iterable $iterable): array
+    private function iterableToArray($iterable): array
     {
+        if (!is_iterable($iterable)) {
+            throw new InvalidArgumentException('Iterable is expected, got ' . gettype($iterable));
+        }
+
         return $iterable instanceof \Traversable ? iterator_to_array($iterable) : (array)$iterable;
     }
+
+    /**
+     * @param $key
+     */
+    private function validateKey($key): void
+    {
+        if (!\is_string($key)) {
+            throw new InvalidArgumentException('Invalid key value.');
+        }
+    }
+
+    /**
+     * @param array $keys
+     */
+    private function validateKeys(array $keys): void
+    {
+        foreach ($keys as $key) {
+            $this->validateKey($key);
+        }
+    }
+
+    /**
+     * @param array $values
+     */
+    private function validateKeysOfValues(array $values): void
+    {
+        $keys = array_map('strval', array_keys($values));
+        $this->validateKeys($keys);
+    }
+
+    /**
+     * Normalizes keys returned from apcu_fetch in multiple mode. If one of the keys is an integer (123) or a string
+     * representation of an integer ('123') the returned key from the cache doesn't equal neither to an integer nor a
+     * string ($key !== 123 and $key !== '123'). Coping element from the returned array one by one to the new array
+     * fixes this issue.
+     * @param array $values
+     * @return array
+     */
+    private function normalizeAPCUoutput(array $values): array
+    {
+        $normalizedValues = [];
+        foreach ($values as $key => $value) {
```

Comment:
It does nothing. Why having it?

Reply:
When creating a new element of the array it casts a key to the string or integer, it's standard PHP behavior, so it normalizes keys of the array.

Reply:
We should add comment explaining it. It's not obvious.